src/main/kotlin/commands/BanCommand.kt

```diff
@@ -0,0 +1,72 @@
+package commands
+
+import Command
+import ConfigManager.readConfigSafe
+import ConfigType
+import Main
+import Permissions.hasPermission
+import Send.error
+import UserConfig
+import arg
+import doesLater
+import greedyString
+import integer
+import net.ayataka.kordis.entity.user.User
+import net.ayataka.kordis.exception.MissingPermissionsException
+import string
+
+object BanCommand : Command("ban") {
+    init {
+        string("user") {
+            integer("deleteMessageDays") {
+                greedyString("reason") {
+                    doesLater { context ->
+                        if (!message.hasPermission(PermissionTypes.COUNCIL_MEMBER)) {
+                            return@doesLater
+                        }
+                        val username: String = context arg "user"
+                        val serverId = readConfigSafe<UserConfig>(ConfigType.USER, false)?.primaryServerId
+                        val user: User = Main.client?.servers?.find(serverId ?: 573954110454366214)?.members?.findByName(username) ?: //username
+                        Main.client?.servers?.find(serverId ?: 573954110454366214)?.members?.find { it.nickname == username } ?: //nick
+                        Main.client?.servers?.find(serverId ?: 573954110454366214)?.members?.find(username.toLong()) ?: //id
+                        run {
+                            message.error("User not found!")
+                            return@doesLater
+                        }
+                        val messageDays: Int = context arg "deleteMessageDays"
+                        val fixedDays = if (messageDays >= 7) 7 else messageDays
+                        val reason: String = context arg "reason"
+                        val fixedReason =
+                            if (reason.isEmpty()) readConfigSafe<UserConfig>(ConfigType.USER, false)?.defaultReason
+                                ?: "No Reason Specified" else reason
+                        if (user.id == message.author?.id) {
```

Comment:
You should also be checking if the person you're trying to ban has COUNCIL_MEMBER permission

Reply:
check line 60, kordis checks the permission. It can't ban a person who have permission

Reply:
It shouldn't be doing that

Reply:
```
net.ayataka.kordis.exception.MissingPermissionsException: Server: testRoom-01 (735248256073990165), User: sourTaste000#7277 (743547656793489488)
```

Reply:
That means their role is above the bots, not ban permission. You should change the message and check council permission.